torch/_utils_internal.py

```diff
@@ -75,7 +75,30 @@ def _functionalize_sync(t):
 
     if isinstance(t, FunctionalTensor):
         maybe_functional_mode = torch._C._get_dispatch_mode(
-            torch._C.TorchDispatchModeKey.FUNCTIONAL
+            torch._C._TorchDispatchModeKey.FUNCTIONAL
+        )
+        # If a FunctionalTensorMode is active while syncing, we don't want it to intercept any ops that get called
+        # when we sync our inner tensor.
+        if maybe_functional_mode is not None:
```

Comment:
This is annoying. The problem I ran into when I added this a few days ago was that:

(1) If there are input mutations in the graph, then they will be re-applied during AOTAutograd when we call `_sync()` from inside of our functionalization kernels.

(2) `_sync()` causes us to regenerate our updated the tensor from the updated base, which dispatches to a bunch of view ops

(3) The input to these view ops is our inner `FunctionalTensorWrapper` (since the sync was called from C++), not the python `FunctionalTensor`

(4) if a python `FunctionalTensorMode` is active, it will complain when it intercepts the view op, since it will see an input that is a C++ `FunctionalTensorWrapper` (aka a normal `torch.Tensor`) instead of a python `FunctionalTensor).

Reply:
Put this comment in the code?

Reply:
Intuitively, the logic here makes sense, because you are going "below" the Python FunctionalTensor, so logically you need to disable the mode before you get there. Because `_functionalize_sync` is not a dispatched operator, this didn't get done automatically for you, so you have to do it here.

Reply:
agreed - will do